workflows/sampletracking.nf

```diff
@@ -40,12 +43,78 @@ workflow SAMPLETRACKING {
     def ch_multiqc_files = Channel.empty()
     def ch_pool_multiqc_files = Channel.empty()
 
+
+    def (ch_sample, ch_snp, ch_rest) = ch_samplesheet
+        .multiMap {meta, sample_bam, sample_bam_index, snp_fastq, snp_bam, snp_bam_index ->
+            sample: [meta, (sample_bam instanceof List && sample_bam.isEmpty() ? null : sample_bam), (sample_bam_index ?: null)]
+            snp   : [meta, (snp_bam   instanceof List && snp_bam.isEmpty()   ? null : snp_bam),     (snp_bam_index   ?: null)]
+            rest  : [meta, (snp_fastq instanceof List && snp_fastq.isEmpty() ? null : snp_fastq)]
+        }
+
+    def (ch_sample_with_idx, ch_sample_no_idx) = ch_sample.branch {
+        _meta, _sample_bam, sample_bam_index ->
+            with_index: sample_bam_index != null
+            no_index  : sample_bam_index == null
+    }
+
+    ch_sample_no_idx = ch_sample_no_idx.filter { _meta, bam, _idx -> bam != null }
```

Comment:
```suggestion

```
This can be skipped since `sample_bam` is a required entry of the pipeline as you can see in this [line](https://github.com/nf-cmgg/sampletracking/blob/fe4bbd74597fe587a4c7550681d47ceb1d91e566/assets/schema_input.json#L70)